plasma-mode.md

```diff
@@ -0,0 +1,203 @@
+# Purpose
+
+This document describes the design of Plasma Mode for different DA Layers with minimal op-node & specs repo changes.
+Once ratified, we will implement the described design.
+
+# Problem Statement + Context
+
+Plasma Mode is a modification the OP Stack protocol where the data is posted to a dedicated DA layer instead of Ethereum. This is done to reduce the cost of operating the specific chain. To maintain security, we provide a mechanism to challenge commitments and make data available on Ethereum when there is a dispute over the commitment.
+
+
+We have the following requirements for Plasma Mode:
+- Data is posted to a DA layer instead of Ethereum
+- There is a DA Challenge which validates that the data is available
+- Optionally, the DA challenge process makes the data available on Ethereum
+- The op-node is able to mark L2 as finalized when commitments used to derive the L2 chain are no longer able to marked as invalid
+
+Additionally, we take the following development-process requirement:
+- The op-node and specification repo does not need to be modified for each new DA Layer
+
+The largest challenge to this development process requirement is that DA Providers have different ways of committing to data and semantics of challenges. We need to create a design that generically supports a broad set of DA Providers without heavy modification or forking of the OP Stack.
+
+## High Level Overview of Plasma Mode
```

Comment:
This and the next section would benefit from a few diagrams to illustrate the main flows, namely posting data to the DA Layer, performing derivation for a commitment and handling challenges.